v2/kafka-to-gcs/src/main/java/com/google/cloud/teleport/v2/templates/KafkaToGCS.java

```diff
@@ -0,0 +1,196 @@
+ * -Dbase-container-image=${BASE_CONTAINER_IMAGE} \
+ * -Dbase-container-image.version=${BASE_CONTAINER_IMAGE_VERSION} \
+ * -Dapp-root=${APP_ROOT} \
+ * -Dcommand-spec=${COMMAND_SPEC}
+ *
+ * # Create an image spec in GCS that contains the path to the image
+ * {
+ *    "docker_template_spec": {
+ *       "docker_image": $TARGET_GCR_IMAGE
+ *     }
+ *  }
+ *
+ * # Execute template:
+ * API_ROOT_URL="https://dataflow.googleapis.com"
+ * TEMPLATES_LAUNCH_API="${API_ROOT_URL}/v1b3/projects/${PROJECT}/templates:launch"
+ * JOB_NAME="kafka-to-gcs-`date +%Y%m%d-%H%M%S-%N`"
+ *
+ * time curl -X POST -H "Content-Type: application/json"     \
+ *     -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+ *     "${TEMPLATES_LAUNCH_API}"`
+ *     `"?validateOnly=false"`
+ *     `"&dynamicTemplate.gcsPath=${BUCKET_NAME}/path/to/image-spec"`
+ *     `"&dynamicTemplate.stagingLocation=${BUCKET_NAME}/staging" \
+ *     -d '
+ *      {
+ *       "jobName":"'$JOB_NAME'",
+ *       "parameters": {
+ *           "bootstrapServers":"broker_1:9092,broker_2:9092",
+ *           "inputTopics":"topic1,topic2",
+ *           "outputDirectory":"'$BUCKET_NAME/path/to/output-location'",
+ *           "outputFileFormat":"text",
+ *           "outputFilenamePrefix":"output",
+ *           "windowDuration":"5m",
+ *           "numShards":"5"
+ *        }
+ *       }
+ *      '
+ * </pre>
+ */
+public class KafkaToGCS {
+
+  /* Logger for class. */
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaToGCS.class);
+
+  /**
+   * Main entry point for pipeline execution.
+   *
+   * @param args Command line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    KafkaToGCSOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(KafkaToGCSOptions.class);
+
+    run(options);
+  }
+
+  /**
+   * Runs a pipeline which reads message from Kafka and writes it to GCS.
+   *
+   * @param options arguments to the pipeline
+   */
+  public static PipelineResult run(KafkaToGCSOptions options) {
+
+    String outputFileFormat = options.getOutputFileFormat().toUpperCase();
+    LOG.info("Requested File Format is " + outputFileFormat);
+
+    // Call the function to check File Format passed by user.
+    WriteToGCSUtility.checkFileFormatFn(outputFileFormat);
+
+    // Create the pipeline
+    Pipeline pipeline = Pipeline.create(options);
+
+    // Register the coder for pipeline
+    FailsafeElementCoder<KV<String, String>, String> coder =
+        FailsafeElementCoder.of(
+            KvCoder.of(
+                NullableCoder.of(StringUtf8Coder.of()), NullableCoder.of(StringUtf8Coder.of())),
+            NullableCoder.of(StringUtf8Coder.of()));
+
+    CoderRegistry coderRegistry = pipeline.getCoderRegistry();
+    coderRegistry.registerCoderForType(coder.getEncodedTypeDescriptor(), coder);
+
+    List<String> topicsList = new ArrayList<>(Arrays.asList(options.getInputTopics().split(",")));
```

Comment:
What happens when topicsList.size() == 0 ?

Reply:
Caused by: org.apache.kafka.common.errors.TimeoutException: Timeout expired while fetching topic metadata
This exception is thrown. Do you need me to handle this one?

Reply:
I think the InvalidtopicException would be thrown at runtime. Is that correct? If so, we are only looking to make sure that the topicsList is not empty. So doing something like checkArgument(topicsList.size > 0, "inputTopics cannot be an empty string") would be sufficient.